src/app/public/modules/overlay/overlay.service.spec.ts

```diff
@@ -0,0 +1,235 @@
+    let overlay = createOverlay();
+
+    SkyAppTestUtility.fireDomEvent(getAllOverlays().item(0), 'click');
+    app.tick();
+
+    expect(getAllOverlays().item(0)).not.toBeNull();
+
+    overlay.closed.subscribe(() => {
+      overlay = createOverlay({
+        disableClose: false
+      });
+
+      SkyAppTestUtility.fireDomEvent(getAllOverlays().item(0), 'click');
+      app.tick();
+
+      expect(getAllOverlays().item(0)).toBeNull();
+
+      overlay.close();
+    });
+
+    overlay.close();
+  }));
+
+  it('should prevent body scroll after another overlay is closed', async(() => {
+    const adapter = TestBed.get(SkyOverlayDomAdapterService);
+    const adapterSpy = spyOn(adapter, 'releaseBodyScroll').and.callThrough();
+
+    const overlay1 = createOverlay({
+      disableScroll: true
+    });
+
+    const overlay2 = createOverlay({
+      disableScroll: true
+    });
+
+    app.tick();
+
+    overlay2.closed.subscribe(() => {
+      expect(adapterSpy).not.toHaveBeenCalled();
+      adapterSpy.calls.reset();
+
+      overlay1.closed.subscribe(() => {
+        expect(adapterSpy).toHaveBeenCalled();
+      });
+
+      overlay1.close();
+    });
+
+    overlay2.close();
+  }));
+
+  it('should optionally show a backdrop', async(() => {
+    let overlay = createOverlay();
+
+    app.tick();
+
+    let backdropElement = document.querySelector('.sky-overlay-backdrop');
+
+    expect(backdropElement).toBeNull();
+
+    overlay.closed.subscribe(() => {
+      overlay = createOverlay({
+        showBackdrop: true
+      });
+
+      app.tick();
+
+      backdropElement = document.querySelector('.sky-overlay-backdrop');
+
+      expect(backdropElement).not.toBeNull();
+
+      overlay.close();
+    });
+
+    overlay.close();
+  }));
+
+  it('should close on navigation change', async(() => {
+    const router = TestBed.get(Router);
+    let overlay = createOverlay();
+
+    app.tick();
+
+    expect(getAllOverlays().item(0)).not.toBeNull();
+
+    router.navigate(['/']);
+    app.tick();
+
+    expect(getAllOverlays().item(0)).toBeNull();
+
+    overlay.close();
+  }));
+
+  it('should optionally remain open on navigation change', async(() => {
+    const router = TestBed.get(Router);
+    let overlay = createOverlay({
+      closeOnNavigation: false
+    });
+
+    app.tick();
+
+    expect(getAllOverlays().item(0)).not.toBeNull();
+
+    router.navigate(['/']);
+    app.tick();
+
+    expect(getAllOverlays().item(0)).not.toBeNull();
+
+    overlay.close();
+  }));
+
+  it('should pass providers to the overlay content', async(() => {
+    const config: SkyOverlayConfig = {
+      providers: [{
+        provide: OverlayFixtureContext,
+        useValue: new OverlayFixtureContext(++uniqueId)
+      }]
+    };
+
+    const overlay = createOverlay(config);
+
+    app.tick();
+
+    expect(getAllOverlays().item(0).textContent).toContain('Overlay content ID: 1');
+
+    overlay.close();
+
+  }));
+
+});
```

Comment:
Is there a need to test for accessibility here?

Reply:
Maybe not explicitly, but I'd prefer to keep it in just in case something about the overlay component is deemed inaccessible. What do you think?

Reply:
Ha, I thought you were implying there was already an accessibility test. I agree, there should be one. Added!